src/sentry/integrations/vercel/integration.py

```diff
@@ -248,55 +248,56 @@ def update_organization_config(self, data):
-            ]
-
-            secrets = []
-            for name, val in zip(secret_names, values):
-                secrets.append(vercel_client.create_secret(vercel_project_id, name, val))
+            env_var_map = {
+                "SENTRY_ORG": {"type": "plain", "value": sentry_project.organization.slug},
+                "SENTRY_PROJECT": {"type": "plain", "value": sentry_project.slug},
+                dsn_env_name: {"type": "plain", "value": sentry_project_dsn},
+                "SENTRY_AUTH_TOKEN": {"type": "secret", "value": sentry_auth_token},
+                "VERCEL_GIT_COMMIT_SHA": {"type": "system", "value": "VERCEL_GIT_COMMIT_SHA"},
+            }
 
-            secrets.append("")
-            for secret, env_var in zip(secrets, env_var_names):
-                self.create_env_var(vercel_client, vercel_project_id, env_var, secret)
+            for env_var, details in env_var_map.items():
+                if details["type"] == "secret":
+                    secret_name = env_var + "_%s" % uuid
+                    details["value"] = vercel_client.create_secret(secret_name, details["value"])
+                self.create_env_var(
+                    vercel_client, vercel_project_id, env_var, details["value"], details["type"]
+                )
 
         config.update(data)
         self.org_integration.update(config=config)
 
-    def env_var_already_exists(self, client, vercel_project_id, name):
-        return any(
-            [
-                env_var
-                for env_var in client.get_env_vars(vercel_project_id)
-                if env_var["key"] == name
-            ]
-        )
+    def create_env_var(self, client, vercel_project_id, key, value, type):
+        data = {
+            "key": key,
+            "value": value,
+            "target": ["production"],
+            "type": type,
+        }
+        try:
+            return client.create_env_variable(vercel_project_id, key, value, data)
+        except ApiError as e:
+            if e.json and e.json.get("error", {}).get("code"):
+                if e.json["error"]["code"] == "ENV_ALREADY_EXISTS":
+                    return self.update_env_variable(client, vercel_project_id, key, value, data)
+            raise
+
+    def update_env_variable(self, client, vercel_project_id, key, value, data):
+        env_var_id = [
+            env_var["id"]
+            for env_var in client.get_env_vars(vercel_project_id)["envs"]
+            if env_var["key"] == key
+        ]
+        if env_var_id:
+            return client.update_env_variable(vercel_project_id, env_var_id[0], data)
 
-    def create_env_var(self, client, vercel_project_id, key, value):
-        if not self.env_var_already_exists(client, vercel_project_id, key):
-            return client.create_env_variable(vercel_project_id, key, value)
-        client.update_env_variable(vercel_project_id, key, value)
+        message = "Could not update environment variable {} in Vercel project {}.".format(
```

Comment:
@ceorourke you should use an f string here now: https://realpython.com/python-f-strings/